R/g.part5.R

```diff
@@ -276,305 +258,283 @@ g.part5 = function(datadir = c(), metadatadir = c(), f0=c(), f1=c(),
             # Add first waking up time, if it is missing:
             ts = g.part5.addfirstwake(ts, summarysleep = summarysleep_tmp2, nightsi, sleeplog, ID,
                                       Nepochsinhour, SPTE_end)
-            if (params_general[["part5_agg2_60seconds"]] == TRUE) { # Optionally aggregate to 1 minute epoch:
-              ts$time_num = floor(as.numeric(iso8601chartime2POSIX(ts$time,tz = params_general[["desiredtz"]])) / 60) * 60
-              
-              # only include angle if angle is present
-              angleColName = ifelse("angle" %in% names(ts), yes = "angle", no = NULL)
-              if (lightpeak_available == TRUE) {
-                ts = aggregate(ts[, c("ACC","sibdetection", "diur", "nonwear", angleColName, "lightpeak", "lightpeak_imputationcode")],
-                               by = list(ts$time_num), FUN = function(x) mean(x))
-              } else {
-                ts = aggregate(ts[,c("ACC","sibdetection", "diur", "nonwear", angleColName)],
-                               by = list(ts$time_num), FUN = function(x) mean(x))
-              }
-              ts$sibdetection = round(ts$sibdetection)
-              ts$diur = round(ts$diur)
-              ts$nonwear = round(ts$nonwear)
-              names(ts)[1] = "time"
-              # # convert back to iso8601 format
-              ts$time = as.POSIXct(ts$time, origin = "1970-1-1", tz = params_general[["desiredtz"]])
-              ws3new = 60 # change because below it is used to decide how many epochs are there in
-              # extract nightsi again
-              time_POSIX = ts$time
-              tempp = as.POSIXlt(time_POSIX) #unclass(time_POSIX)
-              if (is.na(tempp$sec[1]) == TRUE) {
-                time_POSIX = as.POSIXct(ts$time, tz = params_general[["desiredtz"]])
-                tempp = as.POSIXlt(time_POSIX)
-              }
-              sec = tempp$sec
-              min = tempp$min
-              hour = tempp$hour
-              if (params_general[["dayborder"]] == 0) {
-                nightsi = which(sec == 0 & min == 0 & hour == 0)
-              } else {
-                nightsi = which(sec == 0 & min == (params_general[["dayborder"]] - floor(params_general[["dayborder"]])) * 60 & hour == floor(params_general[["dayborder"]])) #shift the definition of midnight if required
+            # Optionally aggregate to part5_epochSizes
+            epochi = unique(params_general[["part5_epochSizes"]])
+            for (ws3new in epochi) {
+              # define classifier to timeuse, fragmentation, or both
+              classifier = c()
+              if (ws3new == params_general[["part5_epochSizes"]][1]) classifier = c(classifier, "timeuse")
+              if (ws3new == params_general[["part5_epochSizes"]][2]) classifier = c(classifier, "fragmentation")
+              if (ws3new != ws3) {
+                ts = aggregate_to_epoch(ts, ws3new, lightpeak_available, params_general)
```

Comment:
You re-use ts from the previous iteration of the loop, which could cause problems:
- If epochi is not ranking from low to high
- First aggregating to, let's say 30 and then to 60 may not be the same as aggregating from 5 to 60 directly.
So, if you do this then I would go back to original ts and aggregate from there.